File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answerList = new ArrayList<>();
        int time = 0;
        int count = 0;
        
        while (progresses.length > 0) {
            if (progresses[0] + time * speeds[0] >= 100) {
                progresses = Arrays.copyOfRange(progresses, 1, progresses.length);
                speeds = Arrays.copyOfRange(speeds, 1, speeds.length);
                count++;
            } else {
                if (count > 0) {
                    answerList.add(count);
                    count = 0;
                }
                time++;
            }
        }
        answerList.add(count);
        
        int[] answer = new int[answerList.size()];
        for (int i = 0; i < answer.length; i++) {
            answer[i] = answerList.get(i);
        }

        return answer;
}
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
# 각 기능 : 진도 100% -> 서비스 반영
# 기능 개발 속도 다름
# 앞에 있는 기능 배포될 때 함께 배포 (뒤에 있는 기능이 먼저 개발되면)

def solution(progresses, speeds):
    answer = []
    index = 0

    while True :
        cnt = 0
        if len(progresses) == sum(answer) :
            break
        for i in range(len(progresses)) :
            progresses[i] += speeds[i]
        if progresses[index] >= 100 :
            cnt += 1
            for j in range(index,len(progresses)) :
                if progresses[j] >= 100 :
                    cnt += 1
                else :
                    break
        if cnt > 0 :
            answer.append(cnt-1)
            index += cnt - 1

    return answer